ui/app/pages/send/send-content/send-amount-row/send-amount-row.component.js

```diff
@@ -26,12 +26,23 @@ export default class SendAmountRow extends Component {
     updateSendAmount: PropTypes.func,
     updateSendAmountError: PropTypes.func,
     updateGas: PropTypes.func,
+    maxMode: PropTypes.bool,
   }
 
   static contextTypes = {
     t: PropTypes.func,
   }
 
+  componentDidUpdate () {
+    const { maxMode, amount, selectedToken, updateSendAmount } = this.props
+
+    if (maxMode && selectedToken) {
+      this.validateAmount(amount)
+      updateSendAmount(amount)
```

Comment:
`updateSendAmount` is already called when `setAmountToMax` is called in `amount-max-button.container.js`, so I think calling it here amounts to calling it a second time.

Maybe there's no reason to call `this.validateAmount` either? 🤔 There can't possibly be an error with the send amount, but it's possible that there might still be an insufficient funds error if the "max" is a very small amount (less than the required gas).

Reply:
Though... `validateAmount` uses `gasTotal`, which hasn't changed at this point. It changes in a subsequent render, after the gas price is re-calculated.

Maybe there should be a separate block for handling `validateAmount`? e.g. if the gas total changes, call `validateAmount` again.